File: src/main/java/io/gitlab/intended/storagenetworks/registry/BlockRegistry.java
package io.gitlab.intended.storagenetworks.registry;

import io.gitlab.intended.storagenetworks.block.*;
import io.gitlab.intended.storagenetworks.item.ModItem;
import net.minecraft.item.BlockItem;
import net.minecraft.util.registry.Registry;

public final class BlockRegistry {

    public static final ModBlock MASTER = new MasterBlock("master");
    public static final ModBlock CRAFTING_TERMINAL = new CraftingTerminalBlock("crafting_terminal");
    public static final ModBlock PROCESSING_TERMINAL = new ProcessingTerminalBlock("processing_terminal");

    public static final ModBlock CABLE = new CableBlock("cable");
    public static final ModBlock STORAGE_CABLE = new StorageCableBlock("storage_cable");
    public static final ModBlock IMPORT_CABLE = new ImportCableBlock("import_cable");
    public static final ModBlock EXPORT_CABLE = new ExportCableBlock("export_cable");
    public static final ModBlock PROCESSING_CABLE = new ProcessingCableBlock("processing_cable");

    public static void register() {
        register(MASTER, CRAFTING_TERMINAL, PROCESSING_TERMINAL, CABLE, STORAGE_CABLE, IMPORT_CABLE, EXPORT_CABLE, PROCESSING_CABLE);
    }

    protected static void register(ModBlock... blocks) {
        for (ModBlock block : blocks) {
            Registry.register(Registry.BLOCK, block.ID, block);
            Registry.register(Registry.ITEM, block.ID, new BlockItem(block, ModItem.getSettings()));
        }
    }

}

File: src/main/kotlin/badasintended/slotlink/block/BlockRegistry.kt
package badasintended.slotlink.block

import badasintended.slotlink.item.ModItem
import net.minecraft.item.BlockItem
import net.minecraft.util.registry.Registry
import net.minecraft.util.registry.Registry.BLOCK
import net.minecraft.util.registry.Registry.ITEM

object BlockRegistry {

    val MASTER = MasterBlock()
    val REQUEST = RequestBlock()

    val CABLE = CableBlock()
    val LINK_CABLE = LinkCableBlock()

    fun init() {
        r(MASTER, REQUEST, CABLE, LINK_CABLE)
    }

    private fun r(vararg modBlocks: ModBlock) {
        for (block in modBlocks) {
            Registry.register(BLOCK, block.id, block)
            Registry.register(ITEM, block.id, BlockItem(block, ModItem.SETTINGS))
        }
    }

}
